lib/commands/owner.js

```diff
@@ -83,13 +85,40 @@ class Owner extends BaseCommand {
     }
   }
 
+  async execWorkspaces ([action, ...args], filters) {
+    await this.setWorkspaces(filters)
+    // ls pkg or owner add/rm package
+    if ((action === 'ls' && args.length > 0) || args.length > 1) {
+      const implicitWorkspaces = this.npm.config.get('workspace', 'default')
+      if (implicitWorkspaces.length === 0) {
+        log.warn(`Ignoring specified workspace(s)`)
+      }
+      return this.exec([action, ...args])
+    }
+    for (const [name] of this.workspaces) {
+      switch (action) {
+        case 'ls':
+        case 'list':
+          return this.ls(name)
```

Comment:
won't the `return` here break out of the loop and only run on the first workspace?

Reply:
They needed to be awaits with breaks after.